### PR TITLE
Adding Fuchsia zxdb debugging walkthrough to docs

### PR DESCRIPTION
**Pull request by andrewpollack**
Adding `zxdb` docs to walkthrough to show debugging steps

### CODE REVIEW

**Comment by andrewpollack** [comment]
r? tmandry

cc @djkoloski

**Comment by andrewpollack** [comment]
r? @tmandry

cc @djkoloski

File: src/doc/rustc/src/platform-support/fuchsia.md
Comment (tmandry) [inline src/doc/rustc/src/platform-support/fuchsia.md:652]: Is the absolute path required? We don't use them elsewhere in the guide

File: src/doc/rustc/src/platform-support/fuchsia.md
Comment (tmandry) [inline src/doc/rustc/src/platform-support/fuchsia.md:738]: The example above shows the user's source code. Does this work automatically, and under what conditions?

File: src/doc/rustc/src/platform-support/fuchsia.md
Comment (tmandry) [inline src/doc/rustc/src/platform-support/fuchsia.md:747]: This should explain why a user would want to check out Fuchsia for this – I assume it's so they can see fdio source – and link to a guide. But then they would need to keep the checkout in sync with their SDK, which I'm sure is possible but not quite sure how to do.

**Comment by tmandry** [comment]
Before landing, please rebase on master and make sure the merge commit is removed. (We generally reserve merge commits for bors and don't allow them in PRs; there's more info on this page: https://rustc-dev-guide.rust-lang.org/git.html#rebasing-and-conflicts.)

File: src/doc/rustc/src/platform-support/fuchsia.md
Comment (andrewpollack) [reply src/doc/rustc/src/platform-support/fuchsia.md:663]: Good point -- tested and absolute path is not necessary. Removing

File: src/doc/rustc/src/platform-support/fuchsia.md
Comment (andrewpollack) [reply src/doc/rustc/src/platform-support/fuchsia.md:749]: The above example shows the user's source code via the `--symbol-path target/x86_64-fuchsia/debug` arg. I'll add an explanation that `--symbol-path` covers user's source code, and that displaying library source code requires `--build-dir` usage

**Review by tmandry — APPROVED** [review]
LGTM after comment

@bors delegate+ rollup

File: src/doc/rustc/src/platform-support/fuchsia.md
Comment (andrewpollack) [reply src/doc/rustc/src/platform-support/fuchsia.md:750]: That makes tons more sense with this background, thank you!

**Pull request merged**
